File: Arrays_and_Strings/javascript/3.js

function urlify(string, trueLength) {
    let url = string.split("")
    for(let i=0; i< trueLength; i++){
        if(url[i] == ' '){
            url[i] = '%20'
        }
    }
    return url.join('');
}

console.log(urlify("Mr John Smith ",13))
console.log(urlify("Mr John  Smith ",14))
console.log(urlify("Mr John Smith ",13))
console.log(urlify("Mr John Smith ",13))
console.log(urlify("Mr John Smith ",13))

File: Arrays_and_Strings/python/3.py

def urlify(string, trueLength):
    url = list(string)
    for i in range(len(url)):
        if url[i] == ' ':
            url[i] = '%20'

    return ''.join(url)


print(urlify("Mr John Smith ", 13))
print(urlify("Mr John  Smith ", 14))
print(urlify("Mr John Smith ", 13))
print(urlify("Mr John Smith ", 13))
print(urlify("Mr John Smith ", 13))
